src/main/java/liquibase/ext/neo4j/lockservice/Neo4jLockService.java

```diff
@@ -0,0 +1,105 @@
+package liquibase.ext.neo4j.lockservice;
+
+import liquibase.database.Database;
+import liquibase.exception.DatabaseException;
+import liquibase.exception.LiquibaseException;
+import liquibase.exception.LockException;
+import liquibase.ext.neo4j.database.Neo4jDatabase;
+import liquibase.lockservice.DatabaseChangeLogLock;
+import liquibase.lockservice.LockService;
+import liquibase.statement.SqlStatement;
+import liquibase.statement.core.RawSqlStatement;
+
+import java.util.Collections;
+import java.util.UUID;
+
+public class Neo4jLockService implements LockService {
+
+    private Database database;
+
+    @Override
+    public int getPriority() {
+        return PRIORITY_DATABASE;
+    }
+
+    @Override
+    public boolean supports(Database database) {
+        return database instanceof Neo4jDatabase;
+    }
+
+    @Override
+    public void setDatabase(Database database) {
+        this.database = database;
+    }
+
+    @Override
+    public void setChangeLogLockWaitTime(long changeLogLockWaitTime) {
+
+    }
+
+    @Override
+    public void setChangeLogLockRecheckTime(long changeLogLocRecheckTime) {
+
+    }
+
+    @Override
+    public boolean hasChangeLogLock() {
+        return true;
+    }
+
+    @Override
+    public void waitForLock() throws LockException {
+
+    }
+
+    @Override
+    public boolean acquireLock() throws LockException {
+        try {
+            // TODO: introduce a RawParameterized statement?
```

Comment:
@nvoxland is there a type in the `SqlStatement` hierarchy that would allow me to run a plain "SQL" statement with parameters?

Reply:
RawSqlStatement would be the closest to what you'd want. It doesn't allow parameters, but lets you just run plain sql.

The reason we don't generally use bind parameters is because we want to support updateSQL style "generate the SQL we can feed into a different CLI" and that isn't going to work with parameters.

Reply:
Makes sense, thanks.